File: MyFamily/app/src/main/java/com/example/myfamily/MainActivity.java
package com.example.myfamily;

import androidx.appcompat.app.AppCompatActivity;

import android.app.Application;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import com.zegocloud.uikit.prebuilt.call.config.ZegoNotificationConfig;
import com.zegocloud.uikit.prebuilt.call.invite.ZegoUIKitPrebuiltCallInvitationConfig;
import com.zegocloud.uikit.prebuilt.call.invite.ZegoUIKitPrebuiltCallInvitationService;

public class MainActivity extends AppCompatActivity {

    EditText edtUsername;
    Button btnProceed;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        edtUsername = findViewById(R.id.edtUsername);
        btnProceed = findViewById(R.id.btnProceed);

        btnProceed.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String username = edtUsername.getText().toString().trim();
                if (!username.isEmpty()){
                    proceedService(username);
                    Intent intent = new Intent(MainActivity.this,CallingActivity.class);
                    intent.putExtra("username",username);
                    startActivity(intent);
                }else {
                    return;
                }
            }
        });
    }

    void proceedService(String userID){
        Application application = getApplication();
        long appID = 1724612326;
        String appSign = "14d50f9b9d11cc5a943a5c387bf150e1a94efe3cfbdf423e3f2d099ecd172f55";
        String userName = userID;

        ZegoUIKitPrebuiltCallInvitationConfig callInvitationConfig = new ZegoUIKitPrebuiltCallInvitationConfig();

        ZegoUIKitPrebuiltCallInvitationService.init(getApplication(), appID, appSign, userID, userName,callInvitationConfig);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        ZegoUIKitPrebuiltCallInvitationService.unInit();
    }
}

File: Stopwatch/app/src/main/java/com/example/stopwatch/MainActivity.kt
package com.example.stopwatch

import android.app.Dialog
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.NumberPicker

class MainActivity : AppCompatActivity() {
    var isRunning = false
    private lateinit var binding: ActivityMainBinding // Declare the binding variable

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater) // Inflate the layout
        setContentView(binding.root)
        binding.imageview.setOnClickListener {
            var dialog = Dialog(this)
            dialog.setContentView(R.layout.dialog)
            var numberpicker=dialog.findViewById<NumberPicker>(R.id.numberpicker)
            numberpicker.minValue=0
            numberpicker.maxValue=5
            dialog.findViewById<<Button>(R.id.set_time).setOnClickListener {
                dialog.dismiss()
            }
            dialog.show()
        }
        binding.run.setOnClickListener {
            if(isRunning){
                isRunning = false

            }
            else{

            }
        }
    }
}